tests/rgen/Microsoft.Macios.Generator.Tests/Classes/Data/Trampolines/ExpectedAVFoundationTrampolinesTrampolines.cs

```diff
@@ -0,0 +1,1369 @@
+	[UserDelegateType (typeof (global::AVFoundation.AVAudioSourceNodeRenderHandlerRaw))]
+	unsafe internal delegate int DAVAudioSourceNodeRenderHandlerRaw (global::System.IntPtr block_ptr, IntPtr isSilence, IntPtr timestamp, uint frameCount, IntPtr outputData);
+
+	/// <summary>This class bridges native block invocations that call into C#</summary>
+	static internal class SDAVAudioSourceNodeRenderHandlerRaw
+	{
+		[Preserve (Conditional = true)]
+		[UnmanagedCallersOnly]
+		[UserDelegateType (typeof (global::AVFoundation.AVAudioSourceNodeRenderHandlerRaw))]
+		internal static unsafe int Invoke (global::System.IntPtr block_ptr, IntPtr isSilence, IntPtr timestamp, uint frameCount, IntPtr outputData)
+		{
+			var del = global::ObjCRuntime.BlockLiteral.GetTarget<global::AVFoundation.AVAudioSourceNodeRenderHandlerRaw> (block_ptr);
+			if (del is not null)
+			{
+				var ret = del (isSilence, timestamp, frameCount, outputData);
+				return ret;
+			}
+			return default;
+		}
+
+		internal static unsafe global::ObjCRuntime.BlockLiteral CreateNullableBlock (global::AVFoundation.AVAudioSourceNodeRenderHandlerRaw? callback)
+		{
+			if (callback is null)
+				return default (global::ObjCRuntime.BlockLiteral);
+			return CreateBlock (callback);
+		}
+
+		[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
+		internal static unsafe global::ObjCRuntime.BlockLiteral CreateBlock (global::AVFoundation.AVAudioSourceNodeRenderHandlerRaw callback)
+		{
+			delegate* unmanaged<global::System.IntPtr, IntPtr, IntPtr, uint, IntPtr, int> trampoline = &Invoke;
+			return new global::ObjCRuntime.BlockLiteral (trampoline, callback, typeof (SDAVAudioSourceNodeRenderHandlerRaw), nameof (Invoke));
+		}
+	}
+	// TODO: generate trampoline for AVFoundation.AVAudioSourceNodeRenderHandlerRaw
+
+	[UnmanagedFunctionPointerAttribute (CallingConvention.Cdecl)]
+	[UserDelegateType (typeof (global::AVFoundation.AVAudioUnitComponentFilter))]
+	unsafe internal delegate byte DAVAudioUnitComponentFilter (global::System.IntPtr block_ptr, global::ObjCRuntime.NativeHandle comp, byte* stop);
+
+	/// <summary>This class bridges native block invocations that call into C#</summary>
+	static internal class SDAVAudioUnitComponentFilter
+	{
+		[Preserve (Conditional = true)]
+		[UnmanagedCallersOnly]
+		[UserDelegateType (typeof (global::AVFoundation.AVAudioUnitComponentFilter))]
+		internal static unsafe byte Invoke (global::System.IntPtr block_ptr, global::ObjCRuntime.NativeHandle comp, byte* stop)
+		{
+			*stop = default;
+			var del = global::ObjCRuntime.BlockLiteral.GetTarget<global::AVFoundation.AVAudioUnitComponentFilter> (block_ptr);
+			if (del is not null)
+			{
+				bool __xamarin_bool__1 = *stop != 0;
+				var ret = del (global::ObjCRuntime.Runtime.GetNSObject<global::AVFoundation.AVAudioUnitComponent> (comp)!, ref __xamarin_bool__1);
+				*stop = __xamarin_bool__1 ? (byte)1 : (byte)0;
+				return ret ? (byte) 1 : (byte) 0;
+			}
+			return default;
```

Comment:
bgen generated:
```cshap
internal static unsafe byte Invoke (IntPtr block, NativeHandle comp, byte* stop) {
	var del = BlockLiteral.GetTarget<global::AVFoundation.AVAudioUnitComponentFilter> (block);
	bool __xamarin_bool__1 = *stop != 0;
	var retval = del ( Runtime.GetNSObject<AVAudioUnitComponent> (comp)!, ref __xamarin_bool__1);
	*stop = __xamarin_bool__1 ? (byte) 1 : (byte) 0;
	return retval ? (byte) 1 : (byte) 0;
}
```
Differences:
1. We are setting the default value of stop. That should be correct in the case the delegate is null.
2. We return the default of byte (0) if the delegate is null.

The init of __xamarin_bool__1 is the same, the assignment of *stop post delegate call too as well as the ret conversion.